hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +264,220 @@ default Table newHmsTable(String hmsTableOwner) {
 
     return newTable;
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  default void commitWithLocking(
+      Configuration conf,
+      BaseMetadata base,
+      BaseMetadata metadata,
+      String baseMetadataLocation,
+      String newMetadataLocation,
+      FileIO io) {
+    boolean newTable = base == null;
+    boolean hiveEngineEnabled = hiveEngineEnabled(conf, metadata);
+    BaseMetastoreTableOperations.CommitStatus commitStatus =
+        BaseMetastoreTableOperations.CommitStatus.FAILURE;
+    boolean updateHiveTable = false;
+    HiveLock lock = lockObject(metadata, conf, catalogName());
+    try {
+      lock.lock();
+      Table tbl = loadHmsTable();
+
+      if (tbl != null) {
+        String tableType = tbl.getTableType();
+        if (!tableType.equalsIgnoreCase(tableType().name())) {
```

Comment:
why do we need to do this?  I feel we should keep existing logic as much as possible.

Reply:
This is needed for view test cases like [this](https://github.com/apache/iceberg/blob/05f99b658189e277bbf32a814ee489cf59add3fa/core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java#L293).

Reply:
Not sure I follow, what happens when we dont have this?  Is it just the error message is wrong?  I feel we can mostly re-use the existing exception (maybe can add the type).

Reply:
@szehon-ho , If we are not adding table existense check before it will throw error as `"Table already exists: ns.view"` or `"View already exists: ns.table"` 
But `ViewCatalogTests` expects error message with exact content Type. 
Say
1. If we are creating table but with same name view already exists, it should be `"View with same name already exists:`
2. If we are creating table with same name table already exists it should be `"Table with same name already exists:`